### PR TITLE
[GR-42598] Native Executable JFR Demo.

### PR DESCRIPTION
**Pull request by LesiaChaban**

### CODE REVIEW

File: native-jfr-demo/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Native Executables with JFR Example
+
+This example demonstrates how JDK Flight Recorder (JFR) as a tool for collecting diagnostic and profiling data running Java application, built into a JVM.
```
Comment (bhoran) [inline native-jfr-demo/README.md:3]:
```suggestion
This example demonstrates how to use the JDK Flight Recorder (JFR) as a tool to collect diagnostic and profiling data in a running Java application, built into a JVM. 
```

File: native-jfr-demo/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Native Executables with JFR Example
+
+This example demonstrates how JDK Flight Recorder (JFR) as a tool for collecting diagnostic and profiling data running Java application, built into a JVM. 
+GraalVM Native Image supports JFR events and users can use [`jdk.jfr.Event`](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Event.html) API with a similar experience to using JFR in the Java HotSpot VM.
```
Comment (bhoran) [inline native-jfr-demo/README.md:4]:
```suggestion
GraalVM Native Image supports JFR events and users can use the [`jdk.jfr.Event`](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Event.html) API with a similar experience to using JFR in the Java HotSpot VM.
```

File: native-jfr-demo/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+To record JFR events when running a native executable, JFR support and JFR recording must be enabled, and this guide covers how to do that.
```
Comment (bhoran) [inline native-jfr-demo/README.md:6]:
```suggestion
To record JFR events when running a native executable, enable JFR support and JFR recording as described in this guide.
```

File: native-jfr-demo/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+> Note: JFR events recording is not supported on GraalVM JDK for Windows. JFR is only supported with native executables built on GraalVM JDK 11.
```
Comment (bhoran) [inline native-jfr-demo/README.md:8]:
```suggestion
> Note: JFR event recording is not supported on GraalVM JDK for Windows. JFR is only supported with native executables built on GraalVM JDK 11.
```
I'm not sure I understand what this means.
Is JFR with GraalVM JDK11 supported on Windows?

File: native-jfr-demo/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Preparation
+
+1. [Download GraalVM](https://www.graalvm.org/downloads/), unzip the archive, set the GraalVM home directory as the `$JAVA_HOME` environment variable and add `$JAVA_HOME/bin` to the `PATH` environment variable.
```
Comment (bhoran) [inline native-jfr-demo/README.md:12]:
```suggestion
1. [Download GraalVM](https://www.graalvm.org/downloads/), unzip the archive, set the `$JAVA_HOME` environment variable as your GraalVM home directory  and add `$JAVA_HOME/bin` to your `PATH` environment variable.
```

File: native-jfr-demo/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
```
Comment (bhoran) [inline native-jfr-demo/README.md:13]: Neither of these might be true. For example, I would expect many users to install GraalVM in a non-user location.

File: native-jfr-demo/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    ``` 
+
+3. Download or clone the demos repository:
+    ```
+    git clone https://github.com/graalvm/graalvm-demos
+    ```
+
+## Build and Run Native Executables with JFR
+
+1. Change directory to the demo subdirectories: _native-jfr-demo_:
```
Comment (bhoran) [inline native-jfr-demo/README.md:37]:
```suggestion
1. Change directory to the demo subdirectory: _native-jfr-demo_:
```

File: native-jfr-demo/README.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    ```
+    cd graalvm-demos/native-jfr-demo
+    ```
+2. Compile the Java file on GraalVM JDK:
```
Comment (bhoran) [inline native-jfr-demo/README.md:41]:
```suggestion
2. Compile the Java file using the GraalVM JDK:
```

File: native-jfr-demo/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    ```bash
+    javac JFRDemo.java
+    ```
+    It creates two class files: _JFRDemo$HelloWorldEvent.class_ and _JFRDemo.class_.
+
+3. Build a native executable with the VM inspection enabled:
```
Comment (bhoran) [inline native-jfr-demo/README.md:47]:
```suggestion
3. Build a native executable with VM inspection enabled:
```

File: native-jfr-demo/README.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    ```bash
+    native-image --enable-monitoring=jfr JFRDemo
+    ```
+    The `--enable-monitoring=jfr` option enables features such as JFR that can be used to inspect the VM.
+
+4. Run the executable and start recording:
+    ```bash
+    ./jfrdemo -XX:+FlightRecorder -XX:StartFlightRecording="filename=recording.jfr"
+    ```
+    It runs the application as a native executable. The `-XX:StartFlightRecording` option enables the built-in Flight Recorder and starts recording to a specified binary file, _recording.jfr_.
```
Comment (bhoran) [inline native-jfr-demo/README.md:57]:
```suggestion
    This command runs the application as a native executable. The `-XX:StartFlightRecording` option enables the built-in Flight Recorder and starts recording to a specified binary file, _recording.jfr_.
```

File: native-jfr-demo/README.md (hunk 6 above)
Comment (bhoran) [reply native-jfr-demo/README.md:13]: I don't think these lines (14-22) are required. We can assume the reader understands the instructions without further details.

File: native-jfr-demo/README.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    export JAVA_HOME=/Users/${current_user}/path/to/graalvm/Contents/Home
+    export PATH=$JAVA_HOME/bin:$PATH
+    ```
+
+2. Install [Native Image](https://www.graalvm.org/dev/reference-manual/native-image/#install-native-image) and VisualVM by running:
+    ```bash
+    gu install native-image jvisualvm
+    ``` 
+
+3. Download or clone the demos repository:
```
Comment (bhoran) [inline native-jfr-demo/README.md:30]:
```suggestion
3. Download the demos repository or clone it as follows:
```

File: native-jfr-demo/README.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+6. Go to **File**, then **Add JFR Snapshot**, browse _recording.jfr_, and open       the selected file. Confirm the display name and click **OK**. Once opened, there is a bunch of options you can check: Monitoring, Threads, Exceptions, etc., but you should be mostly interested in the events browsing. It will look something like this:
+    
+    ![JDK Flight Recorder](img/jfr.png)
+
+    Alternatively, you can view the contents of the recording file in the console window by running this command:
+    ```shell
+   jfr print recording.jfr
+    ```
+    It prints all events recorded by Flight Recorder.
```
Comment (bhoran) [inline native-jfr-demo/README.md:72]:
```suggestion
    It prints all the events recorded by Flight Recorder.
```

**Comment by olyagpl** [comment]
Approved.

File: native-jfr-demo/README.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
+# Recording JFR events with Native Image
+
+This example demonstrates how to use the JDK Flight Recorder (JFR) as a tool to collect diagnostic and profiling data in a running Java application, built into a JVM.
```
Comment (fniephaus) [inline native-jfr-demo/README.md:3]: This example shows how to create a custom JFR event and use that in a native executable. Maybe we could start this guide with an example that shows what standard JFR events are emitted by, for example, `jwebserver`? What do you think?

**Review by fniephaus — CHANGES_REQUESTED** [review]

File: .github/workflows/native-jfr-demo.yml [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          native-image-job-reports: 'true'
+          native-image-pr-reports: 'true'
+      - name: Run 'native-jfr-demo'
+        run: |
+          gu install visualvm
+          cd native-jfr-demo
+          javac JFRDemo.java
+          native-image --enable-monitoring=jfr JFRDemo
+          ./jfrdemo -XX:+FlightRecorder -XX:StartFlightRecording="filename=recording.jfr"
```
[the remark this answers is not in the document]
Comment (LesiaChaban) [reply .github/workflows/native-jfr-demo.yml:38]: So, I need change this command to `./jfrdemo -XX:StartFlightRecording="filename=recording.jfr"`?

[the remark this answers is not in the document]
Comment (fniephaus) [reply .github/workflows/native-jfr-demo.yml:38]: It doesn't hurt but I'd probably remove it, yes.

File: native-jfr-demo/README.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
+# Recording JFR events with Native Image
+
+This demo shows how to create a custom JDK Flight Recorder (JFR) event and use that in a native executable.
+JFR is a tool to collect diagnostic and profiling data in a running Java application, built into a JVM.
```
Comment (fniephaus) [inline native-jfr-demo/README.md:4]:
```suggestion
This demo shows how to create a custom JDK Flight Recorder (JFR) event and use it in a native executable.
JFR is a tool to collect diagnostic and profiling data in a running Java application, and built into the JVM. 
```

File: native-jfr-demo/README.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
+# Recording JFR events with Native Image
+
+This demo shows how to create a custom JDK Flight Recorder (JFR) event and use that in a native executable.
+JFR is a tool to collect diagnostic and profiling data in a running Java application, built into a JVM. 
+GraalVM Native Image supports JFR events and users can use the [`jdk.jfr.Event`](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Event.html) API with a similar experience to using JFR in the JVM. For example, to emit standard JFR events for the `jwebserver` application, you would run:
```
Comment (fniephaus) [inline native-jfr-demo/README.md:5]: We no longer support JDK11:
```suggestion
GraalVM Native Image supports JFR events and users can use the [`jdk.jfr.Event`](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.jfr/jdk/jfr/Event.html) API with a similar experience to using JFR in the JVM. For example, to emit standard JFR events for the `jwebserver` application, you would run:
```

File: native-jfr-demo/README.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+
+3. Build a native executable with VM inspection enabled:
+    ```bash
+    $JAVA_HOME/bin/native-image --enable-monitoring=jfr JFRDemo
+    ```
+    The `--enable-monitoring=jfr` option enables features such as JFR that can be used to inspect the VM.
+
+4. Run the executable and start recording:
+    ```bash
+    ./jfrdemo -XX:StartFlightRecording="filename=recording.jfr"
```
Comment (fniephaus) [inline native-jfr-demo/README.md:55]: You left `-XX:+FlightRecorder` in the `jwebserver` example, so let's use it consistently (it's not needed but also doesn't harm):
```suggestion
    ./jfrdemo -XX:+FlightRecorder -XX:StartFlightRecording="filename=recording.jfr"
```

File: native-jfr-demo/README.md [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+    ```
+    This command runs the application as a native executable. The `-XX:StartFlightRecording` option enables the built-in Flight Recorder and starts recording to a specified binary file, _recording.jfr_.
```
Comment (fniephaus) [inline native-jfr-demo/README.md:57]:
```suggestion
    This command runs the application as a native executable. The `-XX:+FlightRecorder` and `-XX:StartFlightRecording` options enable the built-in Flight Recorder and start recording to a specified binary file, _recording.jfr_.
```

**Review by fniephaus — APPROVED** [review]
LGTM. Please resolve the new conversations before merging. Thanks!

**Pull request merged**